### ISSUE TITLE
Get to know that last login from users

### ISSUE THREAD

**Issue by RyckRichards** [agrodet/Tatoeba-playground#5]
I'm using Tatoeba playground to contact users that may want to translate Tatoeba UI. For now, I'm using [Latest contriutions page](https://tatoeba.org/eng/contributions/latest). However, what if an user haven't logged for the last 10 days, for example, but is willing to help? That would be such a great lost since we have many "partial languages" (languages that were partially translated"

I'd need to look for users who speak a specific language, how fluent he or she is, the other languages he/she speaks and the last time he/she logged in. That'll give me an idea if it worth sending him/her a PM or not

It would be really AWESOME if we it also provided the link for that user like https://tatoeba.org/eng/user/profile/Ricardo14

Thanks a million!

**Comment by agrodet** [comment]
I don't think that Tatoeba should distribute its users last login time (and I don't think they do officially). Therefore, your request seems difficult.  

There might be a way, however to automatize your check to a user's last contribution. I mean by that that we could have something like 
| username | Last contribution |
| --- | --- |
| Ricardo14 | 2020/03/29 |

More than that seems complicated.


> I'd need to look for users who speak a specific language, how fluent he or she is, the other languages he/she speaks

I think that you can achieve this by using the `Users information` notebook. There is a section for natives of a language, and one for all languages of a specific user. Or would like some extension to these two sections? (input more users, having more information, etc?)

> It would be really AWESOME if it also provided the link for that user

This is possible. But it takes some time to build the URL column for many users. I will check in the following days and see what would be a good way to display this information. 

In the meanwhile, could you give me what would be your ideal process? Something like (it's just an example)

1. Choose a language, for example 'jpn'
2. Get all natives with the date of their last contribution
3. Get other languages a specific set of users I choose
4. Display a link to their profile or display a link to the profiles of several users?

If you detail what your ideal use would be, we can imagine more useful functions.

**Comment by RyckRichards** [comment]
> There might be a way, however to automatize your check to a user's last contribution.

That'll work as well. We can deduce that everyone who logged in contribuited to Tatoeba :)

> Or would like some extension to these two sections? (input more users, having more information, etc?)

Well, let's say User12345's native language is French. It would help me if he or she had some knowledge in Dutch or Portuguese for, let's say, "comparative linguistics" (French and Portuguese are romance languages. Dutch isn't.). 

Also, I'm looking for people that speak German but I'd need to know if they speak Spanish, Italian or Japanese. For now I can just "pair" these queries (deu<>spa, deu<>ita, deu<>jpn)

**Comment by agrodet** [comment]
I have added a function called `last_contribution_date` at the bottom of the `Users information` notebook. Could you have a look and tell me if you find it useful? Note that the date displayed is the date of the last "contribution", that is an action related to a sentence. So if someone is active on the Wall or comments but do not touch sentences, it won't really give you the date of the most recent activity.

---

To find all languages spoken by a user, you can use the `languages_of_user` function under the "Languages of a specific user" section. 

> let's say User12345's native language is French. It would help me if he or she had some knowledge in Dutch or Portuguese 

The `natives_speaking_other` function, under the "Natives of X speaking Y" section can give you all French natives speaking Dutch (or Portuguese). 

Using these two functions together, I think you can do lots of things that will ease your work. Let me know how it goes.

**Comment by RyckRichards** [comment]
Indeed. It does help me a looooot! Thanks a lot, Aiji!!!
